### PR TITLE
[doc] Remove the sign of Spark 2.4 support for Batch Read from the Compatibility Matrix table.

### PR DESCRIPTION
**Pull request by Guanpx**
### Docs

Linked issue:  https://github.com/apache/paimon/pull/2998

Remove the sign of Spark 2.4 support  from the Compatibility Matrix table.

### CODE REVIEW

**Comment by Guanpx** [comment]
image

**Comment by Guanpx** [comment]
@zhuangchong

File: docs/content/engines/overview.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ under the License.
 |                                      Flink                                      |  1.15 - 1.19  |     ✅      |      ✅      |      ✅       |  ✅(1.17+)   |        ✅        |       ✅        |        ✅        |
 |                                      Spark                                      |   3.1 - 3.5   |     ✅      |      ✅      |      ✅       |      ✅      |        ✅        |    ✅(3.3+)     |        ✅        |
 |                                      Hive                                       |   2.1 - 3.1   |     ✅      |      ✅      |      ✅       |      ❌      |        ❌        |       ❌        |        ❌        |
-|                                      Spark                                      |      2.4      |     ✅      |      ❌      |      ❌       |      ❌      |        ❌        |       ❌        |        ❌        |
```
Comment (Zouxxyy) [inline docs/content/engines/overview.md:36]: We can just delete this line

Comment (Guanpx) [reply docs/content/engines/overview.md:36]: done
Preview
image

**Comment by JingsongLi** [comment]
+1

**Pull request merged**